File: src/main/java/fr/spoutnik87/musicbot_rest/model/AuditModel.java
package fr.spoutnik87.musicbot_rest.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.io.Serializable;

@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
@Data
public abstract class AuditModel implements Serializable {

  @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/model/AuditModel.kt
package fr.spoutnik87.musicbot_rest.model

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.io.Serializable
import javax.persistence.*

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class AuditModel : Serializable {

    /**
     * Generate a temporary ID. It is overwritten when saved in the repository.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = System.nanoTime()

    @Column(nullable = false, updatable = false)
    @CreatedDate
    val createdAt: Long = 0

    @LastModifiedDate
    val updatedAt: Long = 0
}